tests/unit/nupic/research/spatial_pooler_boost_test.py

```diff
@@ -339,13 +358,86 @@ def boostTestLoop(self, imp):
     self.boostTestPhase3()
     self.boostTestPhase4()
 
+
   def testBoostingPY(self):
-    self.boostTestLoop("py")
+    self.setUp(imp="py")
+    self.boostTestLoop()
 
 
   def testBoostingCPP(self):
-    self.boostTestLoop("cpp")
-
+    self.setUp(imp="cpp")
+    self.boostTestLoop()
+
+
+  def testBoostingDefaultSPParams(self):
+    """Boosting test with SP's default params."""
+    params = {}
+    params['seed'] = SEED
+    # py
+    self.setUp(params, "py") # reinitialize with new params
+    self.boostTestLoop()
+    # cpp
+    self.setUp(params, "cpp")
+    self.boostTestLoop()
+
+
+  def testBoostingSDRPatterns(self):
+    """Boosting test with random patterns, instead of continuous data."""
+    rng = numpy.random.RandomState(seed=SEED)
+    idxA = rng.randint(0, self.inputSize, 20)
+    idxAa = copy.deepcopy(idxA)
+    idxAa[0:10] = rng.randint(0, self.inputSize, 10) # now A and A' have 50% in common
+    idxB = rng.randint(0, self.inputSize, 20) # has (almost) 0% (0.5**20) in common with A (A')
+    idxC = rng.randint(0, self.inputSize, 20)
+    idxD = rng.randint(0, self.inputSize, 20)
+    x = numpy.zeros((5, self.inputSize), dtype=uintType)
+    x[0, idxA] = 1
+    x[1, idxAa] = 1
+    x[2, idxB] = 1
+    x[3, idxC] = 1
+    x[4, idxD] = 1
+    self.setUp(imp="py", data=x)
+    self.boostTestLoop()
+    self.setUp(imp="cpp", data=x)
+    self.boostTestLoop()
+    self.setUp() # set old values
+
+
+  def testBoostingNoDisturbances(self):
+    """Boosting should not create any (significant) anomalies/disturbances."""
+    # This typically happens on simple, periodic data with a bigger SP (num. columns)
+    def runOnce(inp, enc, sp, tp, an):
+      enD = enc.encode(inp)
+      spD = numpy.zeros(sp.getColumnDimensions(), dtype='int32')
+      sp.compute(enD, True, spD)
+      tpD = numpy.zeros(tp.numberOfCols)
+      pastPred = tp.columnConfidences().nonzero()[0] # = predictions at T-1
+      active = spD # active input at T
+      tpD = tp.compute(spD, True, True)
+      anS = an.compute(active, pastPred)
```

Comment:
@scottpurdy are these params correct? In context of the mistake I made in https://github.com/numenta/nupic/issues/2632 
Should we add `def getPredictedColumns() = tp.columnConfidences().nonzero()[0]` just for completeness?